homeassistant/components/bosch_alarm/sensor.py

```diff
@@ -0,0 +1,199 @@
+            identifiers={(DOMAIN, area_unique_id)},
+            name=self._area.name,
+            manufacturer="Bosch Security Systems",
+            model=panel.model,
+            sw_version=panel.firmware_version,
+            via_device=(
+                DOMAIN,
+                unique_id,
+            ),
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Observe state changes."""
+        await super().async_added_to_hass()
+        self.panel.faults_observer.attach(self.schedule_update_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Stop observing state changes."""
+        self.panel.faults_observer.detach(self.schedule_update_ha_state)
+
+
+class FaultingPointsSensor(AreaSensor):
+    """A faults sensor entity for a bosch alarm panel."""
+
+    _attr_translation_key = "faulting_points"
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        super().__init__(panel, area_id, unique_id, "faults")
+
+    @property
+    def native_value(self) -> str:
+        """The state of this faults entity."""
+        return f"{self._area.faults}"
+
+
+class AreaReadyToArmSensor(AreaSensor):
+    """A sensor entity showing the ready state for an area for a bosch alarm panel."""
+
+    _attr_translation_key = "ready_to_arm"
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        super().__init__(panel, area_id, unique_id, "ready_to_arm")
+
+    @property
+    def native_value(self) -> str:
+        """The state of this entity."""
+        if self._area.all_ready:
+            return "home_and_away_ready"
+        if self._area.part_ready:
+            return "home_ready"
+        return "not_ready"
+
+    async def async_added_to_hass(self) -> None:
+        """Observe state changes."""
+        await super().async_added_to_hass()
+        self._area.ready_observer.attach(self.schedule_update_ha_state)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Stop observing state changes."""
+        self._area.ready_observer.attach(self.schedule_update_ha_state)
+
+
+class AreaAlarmsSensor(AreaSensor):
+    """A sensor entity showing the alarms for an area for a bosch alarm panel."""
+
+    _attr_translation_key = "alarms"
+
+    def __init__(self, panel: Panel, area_id: int, unique_id: str) -> None:
+        """Set up a faults sensor entity for a bosch alarm panel."""
+        super().__init__(panel, area_id, unique_id, "alarms")
+
+    @property
+    def icon(self) -> str:
+        """The icon for this alarms entity."""
+        return "mdi:alert-circle-outline"
+
+    @property
+    def native_value(self) -> str:
+        """The state of this alarms entity."""
+        return "\n".join(self._area.alarms) if self._area.alarms else "No Alarms"
```

Comment:
idem

Reply:
I'm not sure that the `binary_sensor` is the best fit for the alarms, as they have some intrinsic structure to them.
There are basically 3 types (burglary, fire, gas) with 3 severity levels (trouble, supervisory, alert). Representing this as 3 `sensors` with 4 possible values seems like a better user experience compared to a big unstructured dump of `binary_sensor`s

Reply:
I also agree with that and was probably going to go that route for alarms

Reply:
But the state shouldn't be all of the things with every type on a new line